src/kakarot/kakarot.cairo

```diff
@@ -152,7 +154,15 @@ func eth_call{
     data: felt*,
 ) -> (return_data_len: felt, return_data: felt*, success: felt) {
     Kakarot.assert_view();
-    return Kakarot.eth_call(origin, to, gas_limit, gas_price, value, data_len, data);
+    let summary = Kakarot.eth_call(origin, to, gas_limit, gas_price, value, data_len, data);
+    if (to == 0) {
+        let (return_data) = alloc();
+        assert [return_data] = summary.address.starknet;
+        assert [return_data + 1] = summary.address.evm;
+        return (2, return_data, 1 - summary.reverted);
```

Comment:
is this return signature compatible with the EVM spec?

Reply:
I don't think there is any spec for this as this return data is a return from EVM to Starknet, so we can put whatever we want.
It was like that actually but we could return the regular return data, which would be in this case the stored bytecode of the deployed account.
This return data is eventually returned nowhere generally speaking (in an execute). We took inspiration from Argent and emit the return data in the EOA, so 
I think that putting the whole deployed bytecode would be expensive for nothing. wdyt?

Reply:
it appears that on remix one is able to read the return_data from the tx itself, though idk if it's done using a receipt or a tx return data directly,
just in general i wonder how it works

for instance in etherjs when one does:
`const current_count = await counter.increment();`

granted increment returns the current counter count, 
how this this handled by the library